02-express-tutorial/app.js

```diff
@@ -1,53 +1,20 @@
 const express = require('express');
 const app = express();
-const port = 3000;
-const { products } = require('./data');
+require('dotenv').config();
+const port = process.env.PORT || 3000;
 
-app.get('/', (req, res) => {
-  res.send('<h1>Home Page</h1><a href="/api/products">products</a>');
-});
-app.get('/api/products', (req, res) => {
-  const newProducts = products.map((product) => {
-    const { id, name, image } = product;
-    return { id, name, image };
-  });
-  res.json(newProducts);
-});
-
-app.get('/api/products/:productID', (req, res) => {
-  const { productID } = req.params;
-  const singleProduct = products.find(
-    (product) => product.id === Number(productID),
-  );
-  if (!singleProduct) {
-    return res.status(404).send('Product Does Not Exist');
-  }
-  return res.json(singleProduct);
-});
+const people = require('./routes/people');
+const auth = require('./routes/auth');
+//static assets
+app.use(express.static('./methods-public'));
+// parse form data;
+app.use(express.urlencoded({ extended: false }));
```

Comment:
Why do we need this? What does it do?

Reply:
app.use(express.static('./methods-public')) - middleware is used to serve static files such as HTML, CSS, JavaScript from the './methods-public' directory when a client makes a request to the server. This middleware eliminates the need to write custom repeated code to handle every static file we want to serve.